### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by emhamza**
### In this milestone I have managed to complete the following tasks:

- [x] Add a column species to animals table
- [x] Insert new data to the animals table
- [x]  Insert diff transactions

**Here is the transaction/queries and their screenshots**

1. Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
space-1

2. Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
space-2

3. Commit the transaction. Verify that changes persist after commit.
space-3

4. Delete all animals born after Jan 1st, 2022.
del-1

5. Create a savepoint for the transaction.
del-2

6. Update all animals' weight to be their weight multiplied by -1.
del-3

7. Rollback to the savepoint
del-4

8. Update all animals' weights that are negative to be their weight multiplied by -1.
del-5

9. Commit transaction
del-6

**Write queries to answer the following questions:**

1. How many animals are there?
q-1

2. How many animals have never tried to escape?
q-2

3. What is the average weight of animals?
q-3

4. Who escapes the most, neutered or not neutered animals?
q-4

5. What is the minimum and maximum weight of each type of animal?
q-5

6. What is the average number of escape attempts per animal type of those born between 1990 and 2000?
q-6

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,5 +7,6 @@ CREATE TABLE animals (
      date_of_birth      date not null,
      escape_attempts    int not null,
      neutered           boolean not null,
-     weight_kg          decimal(10, 2)
+     weight_kg          decimal(10, 2),
+     species             varchar
```
Comment (bobb-Rob) [inline schema.sql]: - [x] You were expected to add a new column called species with a string type using the `ALTER TABLE` command like so 👇🏾 

```sql
ALTER TABLE animals ADD species varchar(255);
```

Please do this instead

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,36 @@ SELECT  name, escape_attempts from animals WHERE   weight_kg > 10.5;
...
 SELECT * from animals WHERE   name <> 'Gabumon';
 SELECT * from animals WHERE  weight_kg >= 10.4 and weight_kg  <= 17.3;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species is null;
+UPDATE animals SET species = 'pokemon' WHERE species is null;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+UPDATE animals SET weight_kg = weight_kg * -1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```
Comment (bobb-Rob) [inline queries.sql:34]: - [x] This commit seems out of place, as it should not be there at all. Looking at the query questions,

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,36 @@ SELECT  name, escape_attempts from animals WHERE   weight_kg > 10.5;
 SELECT * from animals WHERE   neutered = true;
 SELECT * from animals WHERE   name <> 'Gabumon';
 SELECT * from animals WHERE  weight_kg >= 10.4 and weight_kg  <= 17.3;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species is null;
+UPDATE animals SET species = 'pokemon' WHERE species is null;
```
Comment (bobb-Rob) [inline queries.sql]: - [x] Please remove as this is a duplicate code.

Comment (bobb-Rob) [inline queries.sql]: - [x] This highlighted code block should be wrapped in a Transaction block. Showing the before and after like so 👇🏾 

```sql
 SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```
In the code above is just an example pointing to the before and after execution verification. But you will begin with the `BEGIN` keyword, write your query code, then verify before `COMMIT` and after.

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,36 @@ SELECT  name, escape_attempts from animals WHERE   weight_kg > 10.5;
...
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
```
Comment (bobb-Rob) [inline queries.sql]: Some query codes are missing 👇🏾 
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction delete all records in the animals table, then roll back the transaction; After the rollback verify if all records in the animals table still exists.
Take a screenshot of the results of your actions.

- Inside a transaction:
  * `DELETE` all animals born after Jan 1st, 2022
  * Create a savepoint for the transaction
  * `UPDATE` all animals' weight to be their weight multiplied by -1.
  * Rollback to the savepoint
  * `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
  * Commit transaction

### Please add the above missing query codes to the codebase. Please put label to all your query codes like so 👇🏾 
Please Note that the last two queries above are different, hence the should be executed differently with proper labels.

```sql
-- UPDATE the animals table by setting the species column to unspecified, then roll back the change
BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone

-- Add the query question here as label
BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @emhamza, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (emhamza) [reply schema.sql]: Sure thing, I have used the similar SQL but made mistake while adding it in to schema.sql. Thank you so much for clarity! @bobb-Rob 🤙

File: queries.sql (hunk 2 above)
Comment (emhamza) [reply queries.sql:34]: Opps, thank you dear for pointing that out! @bobb-Rob

File: queries.sql (hunk 3 above)
Comment (emhamza) [reply queries.sql]: Definitely!

Comment (emhamza) [reply queries.sql]: I have misinterpreted the instructions, I thought only the main quarries were expected to be there without transaction. But I am glad you give me such progressive feedback! 💯 @bobb-Rob

File: queries.sql (hunk 4 above)
Comment (emhamza) [reply queries.sql]: Thank you once again for such progressive advise, adding labels is a good practice! 🥇  👍  @bobb-Rob

**Review by topeogunleye — APPROVED** [review]
### Hi @emhamza, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Made all changes requested by the previous reviewer. ✔️ 
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**